src/sql/workbench/services/urlBrowser/browser/urlBrowserDialog.ts

```diff
@@ -0,0 +1,450 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import 'vs/css!./media/urlBrowserDialog';
+import { Button } from 'sql/base/browser/ui/button/button';
+import { InputBox } from 'sql/base/browser/ui/inputBox/inputBox';
+import { SelectBox } from 'sql/base/browser/ui/selectBox/selectBox';
+import * as DialogHelper from 'sql/workbench/browser/modal/dialogHelper';
+import { HideReason, Modal } from 'sql/workbench/browser/modal/modal';
+import * as TelemetryKeys from 'sql/platform/telemetry/common/telemetryKeys';
+import { IContextKeyService } from 'vs/platform/contextkey/common/contextkey';
+import { localize } from 'vs/nls';
+import { IContextViewService } from 'vs/platform/contextview/browser/contextView';
+import { attachButtonStyler, attachInputBoxStyler, attachSelectBoxStyler } from 'vs/platform/theme/common/styler';
+import { SIDE_BAR_BACKGROUND } from 'vs/workbench/common/theme';
+import * as DOM from 'vs/base/browser/dom';
+import * as strings from 'vs/base/common/strings';
+import { IClipboardService } from 'sql/platform/clipboard/common/clipboardService';
+import { IThemeService } from 'vs/platform/theme/common/themeService';
+import { ILogService } from 'vs/platform/log/common/log';
+import { ITextResourcePropertiesService } from 'vs/editor/common/services/textResourceConfigurationService';
+import { IAdsTelemetryService } from 'sql/platform/telemetry/common/telemetry';
+import { attachModalDialogStyler } from 'sql/workbench/common/styler';
+import { ILayoutService } from 'vs/platform/layout/browser/layoutService';
+import { Account } from 'azdata';
+import { IAccountManagementService } from 'sql/platform/accounts/common/interfaces';
+import { IAzureAccountService } from 'sql/platform/azureAccount/common/azureAccountService';
+import { azureResource } from 'azurecore';
+import { IAzureBlobService } from 'sql/platform/azureBlob/common/azureBlobService';
+import { Link } from 'vs/platform/opener/browser/link';
+import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { onUnexpectedError } from 'vs/base/common/errors';
+import { Deferred } from 'sql/base/common/promise';
+
+/**
+ * This function adds one year to the current date and returns it in the UTC format.
+ * It's used to pass an expiration date argument to the create shared access signature RPC.
+ * It returns the date in the UTC format for locale time zone independence.
+ * @returns next year's UTC date
+ */
+function nextYear(): string {
+	const today = new Date();
+	const nextYear = new Date(today.getFullYear() + 1, today.getMonth(), today.getDate());
+	return nextYear.toUTCString();
+}
+
+export class UrlBrowserDialog extends Modal {
```

Comment:
The name here is pretty generic - maybe something like `BackupUrlBrowserDialog`?

Reply:
Being verbose is better than being too generic - `BackupRestoreUrlBrowserDialog` is fine.